universal-application-tool-0.0.1/app/services/question/types/DateQuestionDefinition.java

```diff
@@ -0,0 +1,59 @@
+package services.question.types;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+import java.util.Optional;
+import services.Path;
+
+public class DateQuestionDefinition extends QuestionDefinition {
+
+  public DateQuestionDefinition(
+      String name,
+      Path path,
+      Optional<Long> enumeratorId,
+      String description,
+      ImmutableMap<Locale, String> questionText,
+      ImmutableMap<Locale, String> questionHelpText) {
+    super(
+        name,
+        path,
+        enumeratorId,
+        description,
+        questionText,
+        questionHelpText,
+        DateValidationPredicates.create());
+  }
+
+  @AutoValue
+  public abstract static class DateValidationPredicates extends ValidationPredicates {
+
+    public static DateQuestionDefinition.DateValidationPredicates parse(String jsonString) {
+      try {
+        return mapper.readValue(
+            jsonString, AutoValue_DateQuestionDefinition_DateValidationPredicates.class);
+      } catch (JsonProcessingException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    public static DateQuestionDefinition.DateValidationPredicates create() {
+      return new AutoValue_DateQuestionDefinition_DateValidationPredicates();
+    }
+  }
+
+  public DateQuestionDefinition.DateValidationPredicates getDateValidationPredicates() {
+    return (DateQuestionDefinition.DateValidationPredicates) getValidationPredicates();
+  }
+
+  @Override
+  public QuestionType getQuestionType() {
+    return QuestionType.DATE;
+  }
+
+  @Override
+  ImmutableMap<Path, ScalarType> getScalarMap() {
```

Comment:
question (blocking): I know we have been changing the use of paths/scalars recently - is this method now obsolete? If not, could we add the mapping from path to scalar?

Reply:
@kendrickt What do you think we should do here? I know we're planning to remove this at some point, but will things break if we leave it like this in the interim?

Reply:
I'm not really sure if it will break or not if this empty. The only place this is used is csv export, which doesn't really work now that enumerators have been introduced. So I don't think it'll break any more than it is now.

Reply:
I say just leave this for now, and I can clean everything up once I close the csv export ticket.